samples/apply-dictionary-renderer-to-graphics-overlay/src/main/java/com/esri/arcgismaps/sample/applydictionaryrenderertographicsoverlay/components/ApplyDictionaryRendererToGraphicsOverlayViewModel.kt

```diff
@@ -0,0 +1,252 @@
+
+            // Create the point graphics in separate coroutine from a local XML file
+            val graphicsDeferred = viewModelScope.async(Dispatchers.IO) {
+                makeMessageGraphics()
+            }
+
+            val dictionaryRenderer = dictionaryRendererDeferred.await()
+            val pointGraphics = graphicsDeferred.await()
+
+            // sets the graphics in the overlay
+            graphicsOverlay.apply {
+                renderer = dictionaryRenderer
+                graphics.addAll(pointGraphics)
+            }
+
+            // Sets the camera to look a the graphics in the graphics overlay
+            val extent = graphicsOverlay.extent ?: run {
+                messageDialogVM.showMessageDialog(
+                    title = "Error",
+                    description = "The extent of the graphics overlay is null."
+                )
+                return@launch
+            }
+
+            // Set the viewpoint of the scene view using a camera
+            val camera = Camera(
+                lookAtPoint = extent.center,
+                distance = 15_000.toDouble(),
+                heading = 0.toDouble(),
+                pitch = 70.toDouble(),
+                roll = 0.toDouble()
+            )
+            sceneViewProxy.setViewpointCamera(camera)
+        }
+    }
+
+    /**
+     * Create and load a DictionarySymbolStyle from a web style and use it to create a DictionaryRenderer.
+     */
+    private suspend fun createMil2525dDictionaryRenderer(): DictionaryRenderer? {
+        val portalItem =
+            PortalItem(Portal.arcGISOnline(Portal.Connection.Anonymous), "d815f3bdf6e6452bb8fd153b654c94ca")
+
+        val dictionarySymbolStyle = DictionarySymbolStyle(portalItem = portalItem)
+
+        dictionarySymbolStyle.load().onFailure { throwable ->
+            messageDialogVM.showMessageDialog(throwable)
+            return null
+        }
+
+        // Uses the "Ordered Anchor Points" for the symbol style draw rule.
+        dictionarySymbolStyle.configurations.firstOrNull { it.name.equals("model", ignoreCase = true) }
+            ?.let { configuration -> configuration.value = "ORDERED ANCHOR POINTS" }
+
+        return DictionaryRenderer(dictionarySymbolStyle = dictionarySymbolStyle)
+    }
+
+    /**
+     * Create point graphics from a local XML file containing Mil2525d message data.
+     * The XML file is downloaded by the downloader activity.
+     */
+    private fun makeMessageGraphics(): List<Graphic> {
+        val xmlFile = File(provisionPath, "mil2525dmessages.xml")
+        val messageXml = xmlFile.readText()
+        val parser = MessageXmlParser()
+        val messages = parser.parse(messageXml)
+
+        return messages.mapNotNull { message ->
+            val wkid = message.wkid
+            val controlPoints = message.controlPoints
+            if (wkid == null || controlPoints.isEmpty()) {
+                // Optionally log or handle the error here
+                null
```

Comment:
If using above recommendations
```suggestion
                return Result.failure(Exception("Parser failed to retrieve messages."))
```